java/src/main/native/src/TableJni.cpp

```diff
@@ -1110,56 +1112,101 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_readParquet(
   CATCH_STD(env, NULL);
 }
 
+int set_column_metadata(cudf::io::column_in_metadata &column_metadata,
```

Comment:
Nit: These internal functions should be in an anonymous namespace and moved outside of the `extern "C"` block.